components/Msig/CreateConfirm.tsx

```diff
@@ -0,0 +1,201 @@
+  useEffect(() => {
+    if (messageData?.message) {
+      setTimeout(() => {
+        const lCli = new LotusRPCEngine({ apiAddress })
+        lCli
+          .request<Receipt>('StateGetReceipt', { '/': cid }, null)
+          .then((receipt) => setReceipt(receipt))
+          .catch(() => setReceiptError(true))
+      }, 10000)
+    }
+  }, [messageData?.message, apiAddress, cid])
+
+  // Get the safe address after receiving the receipt
+  useEffect(() => {
+    if (receipt) {
+      try {
+        // Verify the exit code
+        if (receipt.ExitCode !== 0) throw new Error('Safe creation failed')
+
+        // Verify the safe address
+        const { robust } = getAddrFromReceipt(receipt.Return)
+        if (!robust) throw new Error('Failed to get address from receipt')
+
+        // Set the safe address
+        setMsigActor(robust)
+      } catch (e) {
+        setCreationError(true)
+      }
+    }
+  }, [receipt, setMsigActor])
+
+  // CID was not provided in query parameters
+  if (!cid) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation status unknown</h2>
+          <hr />
+          <p>Missing CID for monitoring safe creation</p>
+        </ShadowBox>
+      </Dialog>
+    )
+  }
+
+  // Something went wrong while retrieving the message or receipt
+  if (messageError || receiptError) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation status unknown</h2>
+          <hr />
+          <p>Something went wrong</p>
+        </ShadowBox>
+        <ErrorBox>
+          <p>
+            An error occurred while retrieving the status of your safe creation,
+            please refresh the page.
+          </p>
+        </ErrorBox>
+      </Dialog>
+    )
+  }
+
+  // Something went wrong while creating the safe
+  if (creationError) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation failed</h2>
+          <hr />
+          <p>Something went wrong</p>
+        </ShadowBox>
+        <ErrorBox>
+          <p>
+            An error occurred while creating your safe. Click on the CID below
+            to view the transaction in the Glif Explorer:
+          </p>
+          <p>
+            <SmartLink href={`${explorerUrl}/message/?cid=${cid}`}>
+              {cid}
+            </SmartLink>
+          </p>
+        </ErrorBox>
+      </Dialog>
+    )
+  }
+
+  // The safe has been successfully created
+  if (Address) {
+    return (
+      <Dialog>
+        <ShadowBox>
+          <h2>Safe creation completed</h2>
+          <hr />
+          <p>Your new Safe address is:</p>
+          <p>
+            <SmartLink href={`${explorerUrl}/actor/?address=${Address}`}>
+              {Address}
+            </SmartLink>
+          </p>
+        </ShadowBox>
+        <WarningBox>
+          Please make sure to store your Safe address before continuing to
+          prevent losing access to your Safe.
+        </WarningBox>
```

Comment:
Is there any way we can show this as an alert before navigating away from the page or just before refreshing it?